casanovo/denovo/model_runner.py

```diff
@@ -23,307 +25,350 @@
-        strategy=_get_strategy(),
-        val_check_interval=config["every_n_train_steps"],
-    )
-    # Train the model.
-    trainer.fit(
-        model, train_loader.train_dataloader(), val_loader.val_dataloader()
-    )
-    # Clean up temporary files.
-    tmp_dir.cleanup()
+            raise FileNotFoundError("Could not find the model weights file")
+
+        self.model = Spec2Pep().load_from_checkpoint(
+            self.model_filename,
+            **model_params,
+        )
+
+    def initialize_data_module(
+        self,
+        train_index: Optional[AnnotatedSpectrumIndex] = None,
+        valid_index: Optional[AnnotatedSpectrumIndex] = None,
+        test_index: (
+            Optional[Union[AnnotatedSpectrumIndex, SpectrumIndex]]
+        ) = None,
+    ) -> None:
+        """Initialize the data module
+
+        Parameters
+        ----------
+        train_index : AnnotatedSpectrumIndex, optional
+            A spectrum index for model training.
+        valid_index : AnnotatedSpectrumIndex, optional
+            A spectrum index for validation.
+        test_index : AnnotatedSpectrumIndex or SpectrumIndex, optional
+            A spectrum index for evaluation or inference.
+        """
+        try:
+            n_devices = self.trainer.num_devices
+            train_bs = self.config.train_batch_size // n_devices
+            eval_bs = self.config.predict_batch_size // n_devices
+        except AttributeError as err:
+            raise RuntimeError("Please use `initialize_trainer()` first.")
+
+        self.loaders = DeNovoDataModule(
+            train_index=train_index,
+            valid_index=valid_index,
+            test_index=test_index,
+            min_mz=self.config.min_mz,
+            max_mz=self.config.max_mz,
+            min_intensity=self.config.min_intensity,
+            remove_precursor_tol=self.config.remove_precursor_tol,
+            n_workers=self.config.n_workers,
+            train_batch_size=train_bs,
+            eval_batch_size=eval_bs,
+        )
+
+    def _get_index(
+        self,
+        peak_path: str,
+        annotated: bool,
+        msg: str = "",
+    ) -> Union[SpectrumIndex, AnnotatedSpectrumIndex]:
+        """Get the spectrum index.
+
+        If the file is a SpectrumIndex, only one is allowed. Otherwise multiple
+        may be specified.
+
+        Parameters
+        ----------
+        peak_path : str
+            The peak file/directory to check.
+        annotated : bool
+            Are the spectra expected to be annotated?
+        msg : str, optional
+            A string to insert into the error message.
+
+        Returns
+        -------
+        SpectrumIndex or AnnotatedSpectrumIndex
+            The spectrum index for training, evaluation, or inference.
+        """
+        ext = (".mgf", ".h5", ".hdf5")
+        if not annotated:
+            ext += (".mzml", ".mzxml")
+
+        if msg and msg[-1] != " ":
```

Comment:
**nitpick:** Maybe use `strip()` instead to make sure there is no trailing whitespace? And then add a space in each of the longer messages, rather than in this sub-message. Virtually identical effect and this is the simplest piece of code, but somehow it just feels nicer.

Reply:
Absolutely a better solution!